File: lb1/java/WorkerThreadBreaker.java
class WorkerThreadBreaker extends Thread {
    private boolean _canStop = false;
    private int msTimeout = 0;

    public WorkerThreadBreaker(int msTimeout) {
        this.msTimeout = msTimeout;
        this.start();
    }

    public String getMeta() {
        return "Timeout = " + this.msTimeout;
    }

    synchronized public boolean isCanStop() {
        return this._canStop;
    }

    @Override
    public void run() {
        try {
            Thread.sleep(this.msTimeout);
        } catch (InterruptedException error) {
            error.printStackTrace();
        } finally {
            this._canStop = true;
        }
    }
}

File: lb1/c#/lb1c#/WorkerThreadBreaker.cs
namespace lb1 {
    class WorkerThreadBreaker {
        private readonly int _msTimeout = 0;
        private bool _canStop = false;
        private Thread _thread = null;

        public WorkerThreadBreaker(int msTimeout)
        {
            _msTimeout = msTimeout;
            _thread = new Thread(Timeout);
            _thread.Start();
        }

        private void Timeout() {
            try 
            {
                Thread.Sleep(this._msTimeout);
            } 
            catch (ThreadInterruptedException error)
            {
                Console.WriteLine(error.ToString());
            }
            finally
            {
                this._canStop = true;
            }
        }

        public string GetMeta()
        {
            return $"Timeout = {_msTimeout}";
        }

        public bool IsCanStop
        {
            get { lock(this) { return _canStop; } }
        }
    }
}